### DISCUSSION TITLE
rename binary device_class: battery to device_class: battery_alert

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/architecture#737] [General]
followup from a discussion we had on Discord, my proposal is to do as title. That way, `device_class: battery` would be reserved only for entities with a numeric percentage state.

Currently several other entities, (in my config mainly binary_sensors from Doorstrips, Sirens and Floodsensors) for binaries like battery_low (on/off), replace_battery_soon). Also some legacy mobile_app battery_state sensors still have that class.

Also, another icon was suggested for this device_class: battery-heart-variant and battery-alert.

Not sure if this would be breaking, but it would be for the better...

Main benefit would be that a single  device_class would no longer be used for different categories and meanings, which would be a puristic thing. It would also a very practical solution, eg for selecting entities in templates based on their device_class.

bottom line:

device_class: battery is for entities with a numeric percentage (battery_level)
device_class: battery_alert would be for binary entities indicating some type of alerting state

**Comment by Jc2k** [comment]
I'd quite like one specifically for low battery. HomeKit has a characteristic for this so we'd be able to link it through automatically when exporting entities to iOS/Siri/etc.

If it is too generic we won't be able to do that.

**Reply by frenck** [reply]
You can already do this? If the entity is a binary sensor and has the `battery` device class, you could add the characteristic.

**Reply by Jc2k** [reply]
Maybe I was misled by this description:

> ... for binaries like battery_low (on/off), replace_battery_soon) ....

The fact that replace battery was a seperate thing to battery low in this text led me to believe that some devices have a battery fault indicator. If it is documentated that a binary battery device class is ONLY for low battery, that is indeed fine for my use case.

**Reply by Jc2k** [reply]
(And i see that it is :+1:)

**Reply by frenck** [reply]
@Jc2k It is documented as such, in both the end-user and developer documentation:

<https://developers.home-assistant.io/docs/core/entity/binary-sensor#available-device-classes>

**Reply by Mariusthvdb** [reply]
does this also mention the low threshold? Or is that up to the integration to set.

**Reply by Jc2k** [reply]
It's up to the device to decide the low threshold? 5% is an hour for my phone but could be several months for a hue remote.

**Comment by Mariusthvdb** [comment]
well, tbh, my first thought was for a battery_low class, but given the fact several other alert binaries exist, I changed that thought to _alert. I guess it would depend on the implementation to decide when it would be 'on'.

**Comment by frenck** [comment]
> device_class: battery is for entities with a numeric percentage (battery_level)
> device_class: battery_alert would be for binary entities indicating some type of alerting state

I think I'm missing the point for the nature of this change. This separation already exists. Binary Sensors vs Sensors. Each has its own namespace and its own set of device classes. So there is no conflict in device class; They already have a different meaning; And they are already separated...

I'm lost on what this is adding?

**Comment by Mariusthvdb** [comment]
first of all, it's confusing to have an identically named class, do different things.  It's not concise. I feel changing this would optimize the HA code in that way.

Secondly this addition would enable far easier manipulation of entities with set devices in the backend And frontend via templates.

currently, usecase,if we would want the entities with a percentage below a certain threshold, we can not just filter on `device_class: battery` (which would be the single selector required before comparing to threshold)  but have to reject the binaries and possible states with strings. 

Ofc the same could be said the other way round.

lastly, the current meaning of device_class: battery on a binary entity is rather fuzzy .. what does it really mean? it's on or off sure, but there's no real value to that without knowing it's an alert for something. and for what. The icon is not very descriptive either, which, again, doesn't set it apart from a regular battery level icon.

I feel we could solve all of this ambiguity. Maybe I should even rephrase my proposal:

**change** binary_sensor `device_class: battery` to `device_class: battery_alert`.

**Comment by frenck** [comment]
> first of all, it's confusing to have an identically named class, do different things. It's not concise. 

I am strongly disagreeing with that. Both classify the type of information they provide, which is battery information. One is a binary sensor, the other a sensor. Merging the entity platform type into the device class names, will add another layer of complexity that should not be there; especially considering the device class is not designed to tell you anything about the entity platform.


> Secondly this addition would enable far easier manipulation of entities with set devices in the backend And frontend via templates.

Filter on entity platform? I don't see that issue.

> lastly, the current meaning of device_class: battery on a binary entity is rather fuzzy .. what does it really mean? 

<img width="613" alt="image" src="https://user-images.githubusercontent.com/195327/154653279-d15b7ee6-8acc-4b4f-98c3-8524c42da729.png">

<https://www.home-assistant.io/integrations/binary_sensor/>

Based on your last posting, I'm not in favor of this proposal.

**Comment by Mariusthvdb** [comment]
I wasn't trying to merge entity platform into device_class, at least not purposely, and don't think I mentioned that.

My sole purpose is to get rid of the double meaning of the device_class: battery, which is illustrated very simply by 

```
        {{states|selectattr('attributes.device_class','eq','battery')
         |map(attribute='entity_id')|list}}
```
doesn't get clearer than that. A mixed list of battery_levels, low_level, and replace_soon. device_class: battery is not unambiguous perse and needs extra selectors to narrow down the list to get uniquely typed entities.

About the current meaning low/normal. Well its that exactly. What is normal. Is it high, full, semi full. All it does is alert when it is low. Because of that my suggestion would be `battery_alert`, fellow above even suggested `battery_low`.

It's illustrative the other binary battery class has a suffix `_charging`, so it has been done before. 

Normally we say "what's in a name". I hope to have shown there's a lot.

**Comment by frenck** [comment]
> My sole purpose is to get rid of the double meaning of the device_class: battery, which is illustrated very simply by

There is no double meaning. They both provide information on a battery. You are suggesting to mix up that meaning by adding extra meaning to it (the platform), which is a different property.

Therefore I do not agree with this proposal and thus do not support it.

**Comment by Mariusthvdb** [comment]
may I ask what is so different from battery_charging:

from the overview:
battery_charging:  'on' means charging, 'off' means not charging

my proposal:
battery_low:  'on' means low, 'off' means not low

**Reply by frenck** [reply]
I know what you meant with your proposal, and I already stated I didn't agree. I'm not in need for additional explanation. Thanks! 👍

**Comment by frenck** [comment]
Closing this architectural discussion, we are not making the suggested change.

../Frenck
